src/workdayApiHelpers/searchHelpers.ts

```diff
@@ -0,0 +1,163 @@
+  parseSectionDetails,
+} from "../content/utils"
+import { SupplementaryData, ISectionData } from "../content/App/App.types"
+import { defaultColorList } from "../content/Settings/courseColors"
+
+async function buildRequestOptions(code: string, sessionSecureToken?: string) {
+  const urlencoded = new URLSearchParams()
+  urlencoded.append("q", code)
+
+  const headers = new Headers({
+    "Content-Type": "application/x-www-form-urlencoded",
+  })
+
+  if (sessionSecureToken) {
+    urlencoded.append("sessionSecureToken", sessionSecureToken)
+    headers.set("Session-Secure-Token", sessionSecureToken)
+  }
+
+  return {
+    method: "POST",
+    body: urlencoded,
+    redirect: "follow" as RequestRedirect,
+    headers,
+  }
+}
+
+async function fetchSearchData(url: string, requestOptions: RequestInit) {
+  try {
+    const response = await fetch(url, requestOptions)
+    return await response.json()
+  } catch (error) {
+    console.error("Error fetching data:", error)
+    return null
+  }
+}
+
+export async function fetchSecureToken() {
+  const headers = new Headers({
+    Accept: "application/json",
+    "Content-Type": "application/json",
+    "Session-Secure-Token": "",
+  })
+
+  const urlencoded = new URLSearchParams()
+
+  const requestOptions = {
+    method: "POST",
+    body: urlencoded,
+    redirect: "follow" as RequestRedirect,
+    headers,
+  }
+
+  try {
+    const response = await fetch(
+      "https://wd10.myworkday.com/ubc/app-root",
+      requestOptions
+    )
+    const data = await response.json()
+    sessionSecureToken = data["sessionSecureToken"]
+    return sessionSecureToken
+  } catch (error) {
+    console.error("Error fetching secure token:", error)
+    return null
+  }
+}
+
+export async function findCourseId(name: string): Promise<string> {
+  const requestOptions = await buildRequestOptions(name, sessionSecureToken)
+
+  const data = await fetchSearchData(
+    "https://wd10.myworkday.com/ubc/faceted-search2/c12/fs0/search.htmld",
+    requestOptions
+  )
+
+  if (data) {
+    try {
+      const courseId =
+        data["children"][0]["listItems"][0]["title"]["instances"][0][
+          "instanceId"
+        ]
+      return courseId.split("$")[1]
+    } catch (error) {
+      console.error("Error parsing course data:", error)
+      return ""
+    }
+  }
+  return ""
+}
+
+export async function findSupplementaryData(name: string) {
+  const requestOptions = await buildRequestOptions(name, sessionSecureToken)
+
+  const data = await fetchSearchData(
+    "https://wd10.myworkday.com/ubc/faceted-search2/c12/fs0/search.htmld",
+    requestOptions
+  )
+
+  if (data) {
+    try {
+      const path = data["children"][0]["listItems"][0]
+      const instructors = path["detailResultFields"][2]["instances"]
+
+      const instructorsArr: string[] = [""]
+      if (instructors) {
+        for (const item of instructors) {
+          instructorsArr.push(item["text"])
+        }
+      }
```

Comment:
I like the map solution, only issue with why I didnt use it is because instructors has an ``any`` type, so to make TypeScript happy would would have to do ``instructors ? instructors.map((i:any) => i["text"]) : []`` and this fails eslint